examples/client/Locomotion/src/context/newRideContext/index.tsx

```diff
@@ -219,13 +219,19 @@ const RidePageContextProvider = ({ children }: {
       const formattedEstimations = formatEstimations(services, estimations, tags);
       setChosenService(formattedEstimations.find((e: any) => e.eta));
       setServiceEstimations(formattedEstimations);
+      setIsLoading(false);
     } catch (e) {
+      setIsLoading(false);
       setServiceRequestFailed(true);
       setIsReadyForSubmit(false);
-    } finally {
-      setIsLoading(false);
     }
   };
+  useEffect(() => {
+    if (serviceRequestFailed) {
+      changeBsPage(BS_PAGES.ADDRESS_SELECTOR);
+      stopRequestInterval();
+    }
+  }, [serviceRequestFailed]);
```

Comment:
I don't like that we have specific variable for the popup, It should be a popup name